sbi-spec/src/binary.rs

```diff
@@ -829,6 +853,63 @@ impl HartMask {
             hart_id,
         )
     }
+
+    /// Add a hart id to this [HartMark].
+    /// Returns error when hart_id is invalid.
+    #[inline]
+    pub fn insert(&mut self, hart_id: usize) -> Result<(), MaskError> {
+        if self.hart_mask_base == Self::IGNORE_MASK {
+            Ok(())
+        } else if valid_bit(self.hart_mask_base, hart_id) {
+            self.hart_mask |= 1usize << (hart_id - self.hart_mask_base);
+            Ok(())
+        } else {
+            Err(MaskError::InvalidBit)
+        }
+    }
+
+    /// Remove a hart id from this [HartMark].
+    /// Returns error when hart_id is invalid, or it has been ignored.
+    #[inline]
+    pub fn remove(&mut self, hart_id: usize) -> Result<(), MaskError> {
+        if self.hart_mask_base == Self::IGNORE_MASK {
+            Err(MaskError::Ignored)
+        } else if valid_bit(self.hart_mask_base, hart_id) {
+            self.hart_mask &= !(1usize << (hart_id - self.hart_mask_base));
+            Ok(())
+        } else {
+            Err(MaskError::InvalidBit)
+        }
+    }
+
+    /// Returns [HartMaskIter] of self.
+    #[inline]
+    pub const fn iter(&self) -> HartMaskIter {
+        HartMaskIter {
+            inner: HartMask {
+                hart_mask: self.hart_mask,
+                hart_mask_base: self.hart_mask_base,
+            },
+            visited_mask: 0,
+        }
+    }
+}
+
+impl Iterator for HartMaskIter {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
Should we `#[inline]` this function? We can compare compilation results for functions that include or does not include inline hints to see how they differs.

Reply:
With inline hint, it seems even unrolling the loop of `for x in mask.iter()`. 

So I added inline hints here, thanks for this notice.